persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -496,7 +495,21 @@ protected void loginInternally(io.stargate.db.AuthenticatedUser user) {
         @SuppressWarnings("unused")
         Disposable unused =
             request
-                .execute(queryState, queryStartNanoTime)
+                .execute(
+                    queryState.doOnSubscribe(
```

Comment:
This is sort of a nit, but is there any reason why you chain to `queryState`, rather than doing: 
`request.execute().map().doOnSubscribe().subscribe()`?

Reply:
We need to reset thread locals on the _initial_ thread before DSE code starts processing the request. Since the inner RxJava steps (added by DSE) can potentially jump threads, I do not think using the last `doOnSubscribe` for thread locals is guaranteed to be always effective.

This is hypothetical, but since at Stargate level we do not control DSE steps, I thought it would be safer to reset thread locals on the `QueryState` `Single` that is the basic of further processing step.

Hypothetical scenario where this will become critical: `Request.execute()` adds .subscribeOn(TPC.ioScheduler()` to the `QueryState` `Single` (AFAIK something like this may happen in case of Auth cache expiration). In this case the `doOnSubscribe` in front of the final `.subscribe()` call will be invoked on the thread the submits the requests, but code in `QueryMessage.execute()` that deals with tracing will be invoked on the I/O thread.

Reply:
Having thought more on this, I think going back to `doOnEvent` is preferable because it is strongly associated with the flow of the value object, while `onSubscribe` callbacks may be delivered on another thread depending on whether `subscribeOn` steps are inserted before or after it in the pipeline.

Reply:
Ah, I think I finally saw your point @sbtourist . Yes, we can reset thread locals in `doOnSubscribe` just before `.subscribe()`. DSE code (TPCRunnable) should be able to propagate `ExecutorLocals` to other threads as needed... sorry about the confusion. I'll make this change.

Reply:
> We need to reset thread locals on the initial thread before DSE code starts processing the request. Since the inner RxJava steps (added by DSE) can potentially jump threads, I do not think using the last doOnSubscribe for thread locals is guaranteed to be always effective.

AFAIK the `doOnSubscribe()` callback is invoked on the "source" thread, before eventually jumping to the `subscribeOn` thread; I don't have time to double check this right now, I'll get back about it later.